### ISSUE TITLE
Why my pytorch is running on CPU instead of GPU

### ISSUE THREAD

**Issue by RealNewNoob** [pytorch/tutorials#576]
Here are my codes:
**device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")**
print(device)
cuda:0
class AlexNet(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv1 = nn.Conv2d(3, 64, 5, 4, padding = 2) ## in channel, out channel, kernal size, stride, padding
        self.conv2 = nn.Conv2d(64, 192, 5, 1, padding = 2)
        self.conv3 = nn.Conv2d(192, 384, 3, 1, padding = 1)
        self.conv4 = nn.Conv2d(384, 256, 3, 1, padding = 1)
        self.conv5 = nn.Conv2d(256, 256, 3, 1, padding = 1)
        self.fc1 = nn.Linear(256, 120)
    def forward(self,x):
        x = F.relu(self.conv1(x)) 
        x = F.max_pool2d(x,2,2) ## input, kernal size, stride
        x = F.relu(self.conv2(x))
        x = F.max_pool2d(x,2,2)
        x = F.relu(self.conv3(x))
        x = F.max_pool2d(x,2,2)
        x = F.relu(self.conv4(x))
        x = F.max_pool2d(x,2,2)
        x = F.relu(self.conv5(x))
        x = F.max_pool2d(x,2,2)        
        x = x.view(x.size(0), -1)
        x = self.fc1(x)
        return x        
model = AlexNet().to(device)
criterion = nn.CrossEntropyLoss()
optimizer = torch.optim.Adam(model.parameters(), lr = 0.001)

epochs = 50
running_loss_history = []
running_corrects_history = []
test_running_loss_history = []
test_running_corrects_history = []
path = 'play ground 2 save\\'

for e in range(epochs):
    running_loss = 0
    running_corrects = 0
    test_running_loss = 0
    test_running_corrects = 0
    
    for inputs, labels in trainloader:
        labels = labels.type(torch.LongTensor)
        inputs = inputs.to(device)
        labels = labels.to(device)
        outputs = model(inputs)
        loss = criterion(outputs, labels)
        
        optimizer.zero_grad()
        loss.backward()
        optimizer.step()
        
        _, preds = torch.max(outputs, 1)
        running_loss += loss.item()
        running_corrects += torch.sum(preds == labels.data)

    else:
        with torch.no_grad():
            for test_inputs, test_labels in testloader:
                test_labels = test_labels.type(torch.LongTensor)
                test_inputs = test_inputs.to(device)
                test_labels = test_labels.to(device)
                test_outputs = model(test_inputs)
                test_loss = criterion(test_outputs, test_labels)

                _, test_preds = torch.max(test_outputs, 1)
                test_running_loss += test_loss.item()
                test_running_corrects += torch.sum(test_preds == test_labels.data)

        epoch_loss = running_loss/train_size
        epoch_acc = running_corrects.float()/ train_size
        running_loss_history.append(epoch_loss)
        running_corrects_history.append(epoch_acc)
    
        test_epoch_loss = test_running_loss/test_size
        test_epoch_acc = test_running_corrects.float()/ test_size
        test_running_loss_history.append(test_epoch_loss)
        test_running_corrects_history.append(test_epoch_acc)
        print('epoch :', (e+1))
        print('training loss: {:.4f}, acc {:.4f} '.format(epoch_loss, epoch_acc.item()))
        print('validation loss: {:.4f}, validation acc {:.4f} '.format(test_epoch_loss, test_epoch_acc.item()))

**Comment by RealNewNoob** [comment]
torch.cuda.is_available() returns True, and I have used to(device) on both my model and inputs, however, the model is still running on my CPU. Please Help!

**Comment by soumith** [comment]
please use https://discuss.pytorch.org for questions
